src/ralph/networks/tests/test_models.py

```diff
@@ -197,6 +197,25 @@ def test_sub_network_should_assign_automatically(self):
         )
         self.assertEqual(net, subnet.parent)
 
+    def test_sub_network_should_change_automatically(self):
+        net1 = Network.objects.create(
+            name='net', address='10.20.30.0/24'
+        )
+        net2 = Network.objects.create(
+            name='net2', address='10.20.30.240/28'
+        )
+
+        self.assertEqual(net1, net2.parent)
+
+        net3 = Network.objects.create(
+            name='net3', address='10.20.30.128/25'
+        )
+
+        net2 = Network.objects.get(pk=net2.pk)
+        self.assertEqual(net2.parent, net3)
+
+        self.assertEqual(net3.parent, net1)
+
```

Comment:
additional tests:
* add network `10.20.40.128/28` as net4
* change net3 address now to `10.20.40.128/25`
* now net4 should be child of net3, and net2 should be child of net1
* remove net3 - now net4 should be parentless